src/dfuzzer.c

```diff
@@ -928,54 +928,49 @@ void df_print_process_info(int pid)
         char name[PATH_MAX];           // for storing process and package name
         char buf[PATH_MAX + MAXLEN];   // buffer for rpm/dpkg request
         FILE *fp;
-        char *c;
         int fd, stdoutcpy, stderrcpy, ret;
 
 
         sprintf(proc_path, "/proc/%d/exe", pid);
         ret = readlink(proc_path, name, sizeof(name));
         // excludes interprets
-        if (ret != -1 && strstr(name, "python") == NULL && strstr(name, "perl") == NULL)
+        if (ret != -1 && !strstr(name, "python") && !strstr(name, "perl")) {
```

Comment:
`name` isn't terminated here so `valgrind` still complains about it:
```sh
$ sudo valgrind ./dfuzzer -v -n org.freedesktop.resolve1
==32298== Memcheck, a memory error detector
==32298== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==32298== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==32298== Command: ./dfuzzer -v -n org.freedesktop.resolve1
==32298==
Suppressions from './dfuzzer.conf'
[SESSION BUS]
Session bus not found.
[SYSTEM  BUS]
==32298== Conditional jump or move depends on uninitialised value(s)
==32298==    at 0x484F369: strstr (vg_replace_strmem.c:1772)
==32298==    by 0x4032C1: df_print_process_info (dfuzzer.c:937)
==32298==    by 0x4027DE: main (dfuzzer.c:211)
==32298==
==32298== Conditional jump or move depends on uninitialised value(s)
==32298==    at 0x484F372: strstr (vg_replace_strmem.c:1772)
==32298==    by 0x4032C1: df_print_process_info (dfuzzer.c:937)
==32298==    by 0x4027DE: main (dfuzzer.c:211)
==32298==
==32298== Conditional jump or move depends on uninitialised value(s)
==32298==    at 0x484F369: strstr (vg_replace_strmem.c:1772)
==32298==    by 0x403486: df_print_process_info (dfuzzer.c:937)
==32298==    by 0x4027DE: main (dfuzzer.c:211)
==32298==
==32298== Conditional jump or move depends on uninitialised value(s)
==32298==    at 0x484F372: strstr (vg_replace_strmem.c:1772)
==32298==    by 0x403486: df_print_process_info (dfuzzer.c:937)
==32298==    by 0x4027DE: main (dfuzzer.c:211)
==32298==
==32298== Conditional jump or move depends on uninitialised value(s)
==32298==    at 0x484A608: strlen (vg_replace_strmem.c:494)
==32298==    by 0x4C526C7: __vfprintf_internal (in /usr/lib64/libc.so.6)
==32298==    by 0x4C52E8E: buffered_vfprintf (in /usr/lib64/libc.so.6)
==32298==    by 0x4CFF892: __fprintf_chk (in /usr/lib64/libc.so.6)
==32298==    by 0x4034F9: fprintf (stdio2.h:105)
==32298==    by 0x4034F9: df_print_process_info (dfuzzer.c:938)
==32298==    by 0x4027DE: main (dfuzzer.c:211)
==32298==
==32298== Syscall param write(buf) points to uninitialised byte(s)
==32298==    at 0x4CE184F: write (in /usr/lib64/libc.so.6)
==32298==    by 0x4C6569C: _IO_file_write@@GLIBC_2.2.5 (in /usr/lib64/libc.so.6)
==32298==    by 0x4C64A15: new_do_write (in /usr/lib64/libc.so.6)
==32298==    by 0x4C65D6D: _IO_file_xsputn@@GLIBC_2.2.5 (in /usr/lib64/libc.so.6)
==32298==    by 0x4C52EFD: buffered_vfprintf (in /usr/lib64/libc.so.6)
==32298==    by 0x4CFF892: __fprintf_chk (in /usr/lib64/libc.so.6)
==32298==    by 0x4034F9: fprintf (stdio2.h:105)
==32298==    by 0x4034F9: df_print_process_info (dfuzzer.c:938)
==32298==    by 0x4027DE: main (dfuzzer.c:211)
==32298==  Address 0x1ffeffc0f3 is on thread 1's stack
==32298==  in frame #4, created by buffered_vfprintf (???:)
==32298==
[PROCESS: /usr/lib/systemd/systemd-resolved[SYSTEM  BUS]

==32298== Conditional jump or move depends on uninitialised value(s)
==32298==    at 0x484A608: strlen (vg_replace_strmem.c:494)
==32298==    by 0x4C526C7: __vfprintf_internal (in /usr/lib64/libc.so.6)
==32298==    by 0x4C5C653: __vsprintf_internal (in /usr/lib64/libc.so.6)
==32298==    by 0x4CFF5B0: __sprintf_chk (in /usr/lib64/libc.so.6)
==32298==    by 0x4035B4: sprintf (stdio2.h:38)
==32298==    by 0x4035B4: df_print_process_info (dfuzzer.c:1006)
==32298==    by 0x4027DE: main (dfuzzer.c:211)
==32298==
==32302== Syscall param execve(argv[i]) points to uninitialised byte(s)
==32302==    at 0x4CBD96B: execve (in /usr/lib64/libc.so.6)
==32302==    by 0x4CE0331: __spawni_child (in /usr/lib64/libc.so.6)
==32302==    by 0x4CF18E3: clone (in /usr/lib64/libc.so.6)
==32302==  Address 0x1ffefff239 is on thread 1's stack
==32302==
[PACKAGE:
```

Reply:
*d'oh!*, thanks! Even though it was the first thing I wanted to fix, I managed to miss it :-)